build failed with single core

**Describe the bug**

Backend can not be built with a single core `make`, but the build succeeds with `make -j x` (x >2), after the new schemas branch is merged.
It seems that `schemas/layout_schema_2.json.h` isn’t generated before building `Json.cc.o`. 

If we use more than two cores to make, the problem will be gone.

**To Reproduce**
Steps to reproduce the behavior:
1. go to carta-backend/build
2. make clean and make
3. the build will stop at 87% 
`[ 87%] Building CXX object CMakeFiles/carta_backend.dir/src/Util/Json.cc.o
/Users/kchou/bz/carta_build/carta-backend-dev/src/Util/Json.cc:9:10: fatal error: 'schemas/layout_schema_2.json.h' file not found`

**Expected behavior**
Backend can be built with a single core.

**Screenshots or videos**
<img width="1194" alt="Image" src="https://github.com/user-attachments/assets/260bd707-49b2-443f-b7cb-df1ae61ca43e" />

**Platform info (please complete the following information):**
 - OS [e.g. macOS Monterey]: macOS 15.5
 - Browser [e.g. chrome, safari, electron app]: Safari
 - Browser version [e.g. 22]: 
 - Backend branch [e.g. dev, v3b2 release]: dev
 - Frontend branch [e.g. dev, v3b2 release]: dev

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -288,6 +288,9 @@ set(SOURCE_FILES
 
 add_definitions(-DHAVE_HDF5 -DPUGIXML_COMPACT)
 add_executable(carta_backend ${SOURCE_FILES})
+
+add_dependencies(carta_backend carta-schemas)
+
 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     list(REMOVE_ITEM LINK_LIBS uuid)
     target_link_libraries(carta_backend uv ${LINK_LIBS})

